src/ReactDrawer.js

```diff
@@ -24,72 +34,125 @@ class ReactDrawer extends React.Component {
       hiddenOverlay: true,
       hiddenDrawer: true
     };
-    this.handleOperation = this.handleOperation.bind(this);
   }
 
-  handleOperation() {
+  closeDrawer() {
     this.setState({
-      hiddenOverlay: false,
-      hiddenDrawer: false
+      open: false
     });
-
+    if (this.props.onClose) {
+      this.props.onClose();
+    }
+  }
+  openDrawer() {
     this.setState({
-      open: !this.state.open
+      hiddenOverlay: false,
+      hiddenDrawer: false,
+      open: true
     });
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open != this.state.open) {
+      nextProps.open ? this.openDrawer(): this.closeDrawer();
+    }
+  }
+
+  componentDidMount () {
+    this.drawer.addEventListener('webkitAnimationEnd', this.onAnimationEnded);
+  }
+
   componentWillUnmount() {
+    this.drawer.removeEventListener('webkitAnimationEnd', this.onAnimationEnded);
+  }
 
+  getOverlayClassName(theme, animate) {
+    return classNames(
+      'react-drawer-overlay',
+      theme.overlay,
+      animate.animated,
+      {
+        [`${animate.fadeIn}`]: this.state.open,
+        [`${animate.fadeOut}`]: !this.state.open,
+        [`${theme.hidden}`]: this.state.hiddenOverlay
+      }
+    );
   }
 
-  componentWillReceiveProps(nextProps) {
-    if(nextProps.open != this.state.open) {
-      this.handleOperation();
-      this.setState({open: nextProps.open});
-    }else {
-      this.handleOperation();
+  getDrawerClassName(theme, animate) {
+    const position = this.props.position || 'right';
+    const themeAttr = `drawer-${position}`;
+    const drawerTheme = theme[themeAttr];
+    let direction, start;
+    if (this.state.open) {
+      direction = 'In';
+      switch(position) {
+      case 'top':
+        start = 'Down'; break;
+      case 'bottom':
+        start = 'Up'; break;
+      case 'left':
+        start = 'Left'; break;
+      case 'right':
+        start = 'Right'; break;
+      }
+    } else {
+      direction = 'Out';
+      switch(position) {
+      case 'top':
+        start = 'Up'; break;
+      case 'bottom':
+        start = 'Down'; break;
+      case 'left':
+        start = 'Left'; break;
+      case 'right':
+        start = 'Right'; break;
+      }
     }
+    const fade = animate[`fade${direction}${start}`];
```

Comment:
That is a good question, I need to think about it but I think it's a devdependencies.

At the moment animate.css is build within the library so it's still just a dev dependencies (compiled in a css that must be included).

I need to do some improvements on this I will do this in September.